exporter/opentelemetry-exporter-datadog/src/opentelemetry/exporter/datadog/exporter.py

```diff
@@ -282,3 +299,27 @@ def _parse_tags_str(tags_str):
             parsed_tags[key] = value
 
     return parsed_tags
+
+
+def _extract_tags_from_resource(resource):
+    """Parse tags from resource.labels, except service.name which
+    has special significance within datadog"""
+    tags = {}
+    service_name = None
+    if not resource and resource.labels:
+        return [tags, service_name]
+
+    for attribute_key, attribute_value in resource.labels.items():
+        if isinstance(attribute_value, (int, bool, float)):
+            value = str(attribute_value)
+        elif isinstance(attribute_value, str):
+            value = attribute_value
+        else:
+            logger.warning("Could not serialize tag %s", attribute_key)
+            continue
+
```

Comment:
This could be more simply handled by attempting to cast to string inside a `try/except` block and log the exception if casting fails.

Reply:
makes sense, i had ripped it from the zipkin approach but agreed those improvements make sense to me

Reply:
turns out datadog's add span method handles this casting under the hood, so removing this section entirely for perf